src/WebJobs.Script/Description/Node/NodeFunctionInvoker.cs

```diff
@@ -400,16 +401,18 @@ protected override void OnScriptFileChanged(object sender, FileSystemEventArgs e
                     }
                     else if (value is IDictionary<string, object>[])
                     {
-                        value = ((IEnumerable<IDictionary<string, object>>)value).Select(p => NormalizeBindingData(p)).ToArray();
+                        value = ((IEnumerable<IDictionary<string, object>>)value)
+                            .Select(p => NormalizeBindingData(p)).ToArray();
                     }
-                    else
+                    else if (value is IDictionary<string, string>)
                     {
-                        if (!IsEdgeSupportedType(type))
-                        {
-                            // for values not supported by edge, we just
-                            // stringify the value
-                            value = value.ToString();
-                        }
+                        value = ((IDictionary<string, string>)value).ToDictionary(p => Utility.ToLowerFirstCharacter(p.Key), p => p.Value);
```

Comment:
As I expand the metadata in the core bindings, I discover new types we need to support. In this case, the blob metadata had an IDictionary<string, string>. Our coverage should be pretty good now for the range of types that would be used in practice.